eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -32,59 +42,94 @@ function Get-SwaggerInfo()
+            $fileContent = $fileContent -notmatch "csharp: true"
+            $fileContent = $fileContent -notmatch $inputfileRex
+            $fileContent = $fileContent -notmatch "- .*.json"
+
+            $startNum = ($fileContent | Select-String -Pattern '```').LineNumber[0]
+            $configline = ""
+            if ($readme) {
+                Write-Host "Updating autorest.md file to config required readme file."
+                $requirefile = $readme
+                $configline = "require:" + [Environment]::NewLine + "- " + "$requirefile" + [Environment]::NewLine + "csharp: true"
+            } elseif ($inputfile) {
+                Write-Host "Updating autorest.md file to update input-file."
+                if ($inputfile.StartsWith('-')) {
+                    $configline = "input-file:" + [Environment]::NewLine + "$inputfile"
+                } else {
+                    $configline = "input-file:"  + "$inputfile"
+                }
             }
-            if ( $? -ne $True) {
+
+            $fileContent[$startNum - 1] += ([Environment]::NewLine + $configline)
+            $fileContent | Set-Content $autorestFilePath
+            if ( !$? ) {
                 Write-Error "Failed to update autorest.md. exit code: $?"
                 exit 1
             }
-        } elseif ($inputfile -ne "") {
-            Write-Host "Updating autorest.md file to update input-file."
-            $inputfileRex = "input-file *:*"
-            if ((Get-Content $autorestFilePath | Select-String -Pattern $inputfileRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -replace $inputfileRex, ("input-file:" + [Environment]::NewLine + "$inputfile") | Set-Content $autorestFilePath
-            } else {
-                $startNum = (Get-Content $autorestFilePath | Select-String -Pattern '```').LineNumber[0]
-                $fileContent = Get-Content -Path $autorestFilePath
-                $fileContent[$startNum - 1] += ([Environment]::NewLine + "input-file:" + [Environment]::NewLine + "$inputfile")
-                $fileContent | Set-Content $autorestFilePath
+        }
+        
+        # update autorest.md with configuration
+        if ( $autorestConfigYaml) {
+            Write-Host "Update autorest.md with configuration."
+            $range = ($autorestConfigYaml | Select-String -Pattern '```').LineNumber
+            if ( $range.count -gt 1) {
+                $startNum = $range[0];
+                $lines = $range[1] - $range[0] - 1
+                $autorestConfigYaml = ($autorestConfigYaml | Select -Skip $startNum | Select -First $lines) |Out-String
             }
-            
-            if ( $? -ne $True) {
-                Write-Error "Failed to update autorest.md. exit code: $?"
-                exit 1
+
+            Install-Module -Name powershell-yaml -Force -Verbose -Scope CurrentUser
```

Comment:
We should probably figure out if we want to use regex or yaml parsing for updating the file. Based on the settings I've seen I suspect we can just use regex to find/replace the options we need. The examples I showed above should get you most of the way.

Reply:
I may not now exactly which key/pair will be set in the autorestConfig xml. So it may not be able to define the regex to update the autorest.md, so I use yaml parsing to do it.